salt/modules/win_iis.py

```diff
@@ -160,6 +163,37 @@ def _srvmgr(cmd, return_json=False):
     return ret
 
 
+def _collection_match_to_index(pspath, colfilter, name, match):
+    '''
+    Returns index of collection item matching the match dictionary.
+    '''
+    collection = get_webconfiguration_settings(pspath, [{'name': name, 'filter': colfilter}])[0]['value']
+    for idx, collect_dict in enumerate(collection):
+        if all(item in collect_dict.items() for item in match.items()):
+            return idx
+    return -1
+
+
+def _prepare_settings(pspath, settings):
+    '''
+    Prepare settings before execution wit get or set functions.
+    Removes settings with a match parameter when index is not found.
+    '''
+    prepared_settings = []
+    for setting in settings:
+        match = re.search('Collection\[(\{.*\})\]', setting['name'])
```

Comment:
This line is failing lint as well... Might need to add an `r` here. Not sure how that would affect the regex though. You'll need to test.

Reply:
If it must be formatted like this, then you'll need to skip the lint test for this line by adding this to the end:
```
# pylint: disable=W1401
```
There should be two spaces between the end of the code and the `#`

Reply:
The backslashes here are not interpreted by Python in any way ( neither \[ nor \{  means something ). So it is safe to add the raw prefix. I tested it, it works.